### PR TITLE
GTT Orders

### PR DESCRIPTION
**Pull request by rhnvrm**
Adds support for GTT Orders.

### CODE REVIEW

File: gtt.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	TriggerValues []float64 `json:"trigger_values"`
+}
+
+// GTT represents a single GTT order.
+type GTT struct {
+	ID            int          `json:"id"`
+	UserID        string       `json:"user_id"`
+	ParentTrigger interface{}  `json:"parent_trigger"`
+	Type          GTTType      `json:"type" url:""`
+	CreatedAt     string       `json:"created_at"`
```
Comment (vividvilla) [inline gtt.go]: Type should be `.Time` field

File: gtt.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+type Trigger interface {
+	TriggerValues() []float64
+	LimitPrices() []float64
+	Quantities() []float64
+}
+
+type TriggerParams struct {
+	TriggerValue float64
+	LimitPrice   float64
+	Quantity     float64
```
Comment (vividvilla) [inline gtt.go]: Make it as integer.

File: gtt.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	Condition     GTTCondition `json:"condition"`
+	Orders        []Order      `json:"orders"`
+	Meta          GTTMeta      `json:"meta"`
+}
+
+// Trigger is an abstraction over multiple GTT types.
+type Trigger interface {
+	TriggerValues() []float64
+	LimitPrices() []float64
+	Quantities() []float64
```
Comment (vividvilla) [inline gtt.go]: Make it as integer.

File: gtt.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+// GTTParams is a helper struct used to populate an
+// actual GTT before sending it to the API.
+type GTTParams struct {
+	Tradingsymbol   string
+	Exchange        string
+	LastPrice       float64
+	TransactionType string
+	Type            GTTType
```
Comment (vividvilla) [inline gtt.go]: this can be removed

File: gtt.go (hunk 2 above)
Comment (rhnvrm) [reply gtt.go]: orders.go also is using float64

https://github.com/zerodhatech/gokiteconnect/blob/master/orders.go#L36

**Pull request merged**